crates/noirc_frontend/src/hir/scope/mod.rs

```diff
@@ -165,13 +156,14 @@ impl<K: std::hash::Hash + Eq + Clone, V> ScopeForest<K, V> {
             .expect("ice: expected a scope tree, however none was found")
     }
 
-    /// Starting a for loop requires access to the outside scope.
-    /// Once the for loop has finished, we remove it from the scope tree
-    pub fn start_for_loop(&mut self) {
+    /// The beginning of a scope always correlates with the start of a block {}.
+    /// This can be in if expressions, for loops, or functions.
```

Comment:
Merging this into one definitely simplifies things. Does this take into account that the scope for a for loop/if expression is different to that of a function? a for loop can have access to variables defined in the function before the for loop was initialised, whereas a function cannot do this.

Reply:
It does - it uses the same enter and exit scope functions behind the scene that for loops used to use.